clients/src/main/java/org/apache/kafka/clients/admin/internals/AlterShareGroupOffsetsHandler.java

```diff
@@ -0,0 +1,182 @@
+        return AdminApiFuture.forKeys(Collections.singleton(CoordinatorKey.byGroupId(groupId)));
+    }
+
+    @Override
+    public String apiName() {
+        return "alterShareGroupOffsets";
+    }
+
+    @Override
+    public ApiResult<CoordinatorKey, Map<TopicPartition, Errors>> handleResponse(Node broker, Set<CoordinatorKey> keys, AbstractResponse abstractResponse) {
+        AlterShareGroupOffsetsResponse response = (AlterShareGroupOffsetsResponse) abstractResponse;
+        final Map<TopicPartition, Errors> partitionResults = new HashMap<>();
+        final Set<CoordinatorKey> groupsToUnmap = new HashSet<>();
+        final Set<CoordinatorKey> groupsToRetry = new HashSet<>();
+
+        for (AlterShareGroupOffsetsResponseData.AlterShareGroupOffsetsResponseTopic topic : response.data().responses()) {
+            for (AlterShareGroupOffsetsResponseData.AlterShareGroupOffsetsResponsePartition partition : topic.partitions()) {
+                TopicPartition topicPartition = new TopicPartition(topic.topicName(), partition.partitionIndex());
+                Errors error = Errors.forCode(partition.errorCode());
+
+                if (error != Errors.NONE) {
+                    handleError(
+                            groupId,
+                            topicPartition,
+                            error,
+                            partitionResults,
+                            groupsToUnmap,
+                            groupsToRetry
+                    );
+                } else {
+                    partitionResults.put(topicPartition, error);
+                }
+            }
+        }
+
+        if (groupsToUnmap.isEmpty() && groupsToRetry.isEmpty()) {
+            return ApiResult.completed(groupId, partitionResults);
+        } else {
+            return ApiResult.unmapped(new ArrayList<>(groupsToUnmap));
+        }
+    }
+
+    private void handleError(
+            CoordinatorKey groupId,
+            TopicPartition topicPartition,
+            Errors error,
+            Map<TopicPartition, Errors> partitionResults,
+            Set<CoordinatorKey> groupsToUnmap,
+            Set<CoordinatorKey> groupsToRetry
+    ) {
+        switch (error) {
+            case COORDINATOR_LOAD_IN_PROGRESS:
+            case REBALANCE_IN_PROGRESS:
+                log.debug("AlterShareGroupOffsets request for group id {} returned error {}. Will retry.",
+                        groupId.idValue, error);
+                groupsToRetry.add(groupId);
+                break;
+            case COORDINATOR_NOT_AVAILABLE:
+            case NOT_COORDINATOR:
+                log.debug("AlterShareGroupOffsets request for group id {} returned error {}. Will rediscover the coordinator and retry.",
+                        groupId.idValue, error);
+                groupsToUnmap.add(groupId);
+                break;
+            case GROUP_ID_NOT_FOUND:
+            case NON_EMPTY_GROUP:
```

Comment:
The KIP mentions `GROUP_NOT_EMPTY`, probably we need to update KIP as per the Errors.java.
cc: @AndrewJSchofield

Reply:
Quite right. It should be `NON_EMPTY_GROUP`. I'll update the KIP :)